src/electron/routing_service.ts

```diff
@@ -125,18 +131,14 @@ export class RoutingDaemon {
         });
       };
       newSocket.once('error', initialErrorHandler);
-
-      const cleanup = () => {
-        newSocket.removeAllListeners();
-        this.fulfillDisconnect();
-      };
-      newSocket.once('close', cleanup);
-      newSocket.once('error', cleanup);
     });
   }
 
   private dataHandler(data: Buffer) {
-    const message: RoutingServiceResponse = JSON.parse(data.toString());
+    const message = this.parseRoutingServiceResponse(data);
+    if (!message) {
```

Comment:
Should we explicitly check for undefined here?  I guess RoutingServiceResponse will never be false-y so maybe this isn't an issue?

Reply:
Yeah, this is a standard way of checking for undefined.